sympy/ntheory/digits.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import print_function, division
+
+import math
+import re
+
+from collections import Counter
+
+
+def count_digits(n, base=10):
+    """
+    A frequency counter for the digits of an integer ``n`` given in a base
+    ``base``.
+    """
+    base_func = {2: bin, 8: oct, 10: str, 16: hex}
+
+    if base in [2, 8, 10, 16]:
+        s = re.sub(r'0[a-z]{1}', '', base_func[base](n))
+        count = Counter(s)
+        return {int(k, base=16): v for k, v in count.items()}
+
+    m = math.floor(math.log(n, base)) + 1
```

Comment:
It looks like you are trying to find the number of digits here.

Reply:
A better approach can be, `m = math.ceil(math.log(n, base))`.

Reply:
I believe the correct formula for the number of digits in a number `N` in a base `b` is

    floor(log(N, b)) + 1

This function is not the same as `ceil(log(N, b))`, which can produce an underestimate, e.g. consider the base-4 integer `321` (3 * 4^2 + 2 * 4^1 + 1 * 4^0). Using `ceil(log(N, b)` we get

    ceil(log(123, 4)) = 4

Reply:
@czgdp1807 As the `n` is being given in decimal form (the base digits are mapped to decimal integers) it is actually correct here to log 10:

    m = math.floor(math.log10(n)) + 1

e.g. for n = 4236, base = 8.

Reply:
Oh yeah, I got it wrong. Resolving.